snuba/clusters/cluster.py

```diff
@@ -0,0 +1,95 @@
+from typing import MutableMapping, Set
+
+from snuba import settings
+from snuba.clickhouse.native import ClickhousePool, NativeDriverReader
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.reader import Reader
+
+
+class ClickhouseCluster:
```

Comment:
Wait, I think this hierarchy is inverted.

Reply:
Cluster should be the parent class and only expose the abstract concepts like reader and storage set, while the subclass (clickhouse should be used only when you need the clickhouse specific concept). This allows us to drive the design in a direction that only depends on the abstract class (and as a consequence being independent on the implementation).
Was there a reason to make the parent class, clickhouse specific?

Reply:
This was a mistake, inverted these classes now.